src/main/kotlin/com/ampnet/mailservice/service/impl/MailServiceImpl.kt

```diff
@@ -115,27 +115,28 @@ class MailServiceImpl(
         sendEmail(mail)
     }
 
-    private fun createMailMessage(to: List<String>, subject: String, text: String): MimeMessage {
-        val mail = mailSender.createMimeMessage()
-        val helper = MimeMessageHelper(mail)
-        helper.setFrom(applicationProperties.mail.sender)
-        helper.setTo(to.toTypedArray())
-        helper.setSubject(subject)
-        helper.setText(text, true)
-        helper.setSentDate(Date())
-        return mail
+    private fun createMailMessage(to: List<String>, subject: String, text: String): List<MimeMessage> {
+        return to.map {
+            val mail = mailSender.createMimeMessage()
+            val helper = MimeMessageHelper(mail)
+            helper.setFrom(applicationProperties.mail.sender)
+            helper.setTo(it)
+            helper.setSubject(subject)
+            helper.setText(text, true)
+            helper.setSentDate(Date())
+            mail
+        }
     }
 
-    private fun sendEmail(mail: MimeMessage) {
+    private fun sendEmail(mails: List<MimeMessage>) {
         if (applicationProperties.mail.enabled.not()) {
-            logger.warn { "Sending email is disabled. \nEmail: ${mail.content}" }
+            logger.warn { "Sending email is disabled. \nEmail: ${mails.first().content}" }
```

Comment:
Are you sending email to the only first item in the list?

Reply:
No, it's just for logging purposes, the content of each mail is the same.